packages/react-dom-bindings/src/events/ReactDOMEventReplaying.js

```diff
@@ -303,7 +305,17 @@ function attemptExplicitHydrationTarget(
   // Try to unify them. It's a bit tricky since it would require two return
   // values.
   const targetInst = getClosestInstanceFromNode(queuedTarget.target);
-  if (targetInst !== null) {
+  if (targetInst === null) {
+    const reject = queuedTarget.reject;
+    if (reject) {
+      reject(
+        new Error(
+          'Cannot schedule hydration, target is not a React component.',
```

Comment:
This might not be the correct reason all the time I think. It could've been deleted from the tree before being hydrated, e.g. due to a hydration mismatch.

In that case there's no marker on the parent that was removed from the DOM.